tests/MongoDB.Driver.Core.Tests/Core/Misc/SemanticVersionTests.cs

```diff
@@ -40,6 +35,28 @@ public class SemanticVersionTests
         [InlineData("1.0.1", "1.0.0", 1)]
         [InlineData("1.0.0-alpha", "1.0.0-beta", -1)]
         [InlineData("1.0.0-beta", "1.0.0-alpha", 1)]
+        [InlineData("4.4.0-rc12", "4.4.0-rc12", 0)]
+        [InlineData("4.4.0-rc12", "4.4.0", -1)]
+        [InlineData("4.4.1-rc12", "4.4.0", 1)]
+        [InlineData("4.4.0-rc13", "4.4.0-rc12", 1)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.0-489-gb8f58d7", 0)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.1", -1)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.0", 1)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.0-5-g5a9a742f6f", 1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc12-5-g5a9a742f6f", 0)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc13", -1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc12", 1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc12-4-g5a9a742f6f", 1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc13-5-g5a9a742f6f", -1)]
+        [InlineData("4.4.0-alpha1", "4.4.0-alpha", 1)]
+        [InlineData("4.4.0-alpha3", "4.4.0-beta1", -1)]
+        [InlineData("4.4.0-alpha0", "4.4.0-alpha", 1)]
+        [InlineData("4.4.0-alpha2", "4.4.0-alpha1", 1)]
+        [InlineData("4.4.0-alpha2", "4.4.0-alpha11", -1)]
+        [InlineData("4.4.0-alpha", "4.4.0", -1)]
+        [InlineData("4.4.0-alpha", "4.4.0-alpha", 0)]
+        [InlineData("4.4.0-beta", "4.4.0-5-g5a9a742f6f", -1)]
+        [InlineData("4.4.0-alpha12-5-g5a9a742f6f", "4.4.0-rc13-5-g5a9a742f6f", -1)]
         public void Comparisons_should_be_correct(string a, string b, int comparison)
         {
             var subject = SemanticVersion.Parse(a);
```

Comment:
I changed lines 62-65 to:

    var subject = a == null ? null : SemanticVersion.Parse(a);
    var comparand = b == null ? null : SemanticVersion.Parse(b);
    if (subject != null)
    {
        subject.Equals(comparand).Should().Be(comparison == 0);
        subject.CompareTo(comparand).Should().Be(comparison);
    }

I later saw that you did something similar but slightly different in ServerVersionTests.cs.

Pick one or the other and make both approaches the same.

Reply:
Done.